src/Trinity.C/src/Storage/MTHash/MTHash.Resize.cpp

```diff
@@ -18,25 +18,25 @@ namespace Storage
             return;
         }
 
-        if ((CurrentEntryCount + UInt32_Contants::GuardedEntryCount) >= TrinityConfig::MaxEntryCount())
+        if ((CurrentEntryCount + UInt32_Contants::GuardedEntryCount) >= TrinityConfig::ReserveEntriesPerMTHash())
         {
 			//XXX should not exit here
             Trinity::Diagnostics::FatalError(1, "Memory Trunk {0} is out of Memory::", memory_trunk->TrunkId);
         }
 
         uint32_t expanded_entry_count = CurrentEntryCount + UInt32_Contants::EntryExpandUnit;
-        if ((expanded_entry_count + UInt32_Contants::GuardedEntryCount) >= TrinityConfig::MaxEntryCount())
+        if ((expanded_entry_count + UInt32_Contants::GuardedEntryCount) >= TrinityConfig::ReserveEntriesPerMTHash())
         {
-            expanded_entry_count = TrinityConfig::MaxEntryCount() - UInt32_Contants::GuardedEntryCount;
+            expanded_entry_count = (uint32_t) (TrinityConfig::ReserveEntriesPerMTHash() - UInt32_Contants::GuardedEntryCount);
         }
 
         uint64_t size_expanded = expanded_entry_count - CurrentEntryCount;
 		Trinity::Diagnostics::WriteLine(LogLevel::Verbose, "MemoryTrunk {0}: Expand: {1}->{2}", memory_trunk->TrunkId, CurrentEntryCount, expanded_entry_count);
 
-        Memory::ExpandMemoryFromCurrentPosition((char*)CellEntries + ((CurrentEntryCount + UInt32_Contants::GuardedEntryCount) << 3), size_expanded << 3);
-        memset((char*)CellEntries + ((CurrentEntryCount + UInt32_Contants::GuardedEntryCount) << 3), -1, size_expanded << 3);
+        Memory::ExpandMemoryFromCurrentPosition((char*)CellEntries + ((CurrentEntryCount + UInt32_Contants::GuardedEntryCount) * sizeof(CellEntries[0])), size_expanded * sizeof(CellEntries[0]));
+        memset((char*)CellEntries + ((CurrentEntryCount + UInt32_Contants::GuardedEntryCount) * sizeof(CellEntries)), -1, size_expanded * sizeof(CellEntries[0]));
```

Comment:
`sizeof(CellEntries[0])?`

Reply:
Should we define constants for `sizeof(CellEntries[0])`, `sizeof(MTEntries[0])`, and etc.